### ISSUE TITLE
cupsCopyDestInfo fails

### ISSUE THREAD

**Issue by SonnyX** [OpenPrinting/cups#175]
I'm having issues connecting & copying destination info of specific printers...
Both when using a connection that was set up beforehand or by using the default http provider (null)

Considering its not the case for all printers I don't think that I am doing anything in particular wrong, however I believed these printers to be working in the past.

IPP is enabled on these printers. I am using **CUPS 2.3.3op2**

![afbeelding](https://user-images.githubusercontent.com/784475/114694637-07794d00-9d1b-11eb-81b9-e983c7dd2ebf.png)

```c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#include <cups/cups.h>

int main(void) {
  printf("Setting username\n");
  fflush(stdout);
  cupsSetUser("print\0");
  cups_dest_t *dests;
  printf("Getting destinations\n");
  fflush(stdout);
  int amount = cupsGetDests(&dests);
  while (amount == 0) {
    printf("Found %i destinations, retrying in 2 seconds.\n",amount);
    fflush(stdout);
    cupsFreeDests(amount, dests);
    sleep(2);

    printf("Getting destinations\n");
    fflush(stdout);
    amount = cupsGetDests(&dests);
  }
  printf("Found %i destinations\n\n\n", amount);
  fflush(stdout);
  for (int i = amount - 1; i >= 0; i--) {
    printf("Found destination: %s\n", dests[i].name);
    fflush(stdout);
    int cancel = 0;
    http_t *connection = cupsConnectDest(&dests[i], CUPS_DEST_FLAGS_NONE, 30000, &cancel, NULL, 0, NULL, NULL);
    printf("Connection: %x, Error report: 0x%x - %s\n", connection, cupsLastError(), cupsLastErrorString());
    fflush(stdout);
    cups_dinfo_t *dinfo = cupsCopyDestInfo(NULL, &dests[i]);
    printf("Destination Info: %x, Error report: 0x%x - %s\n\n\n", dinfo, cupsLastError(), cupsLastErrorString());

    fflush(stdout);
  } 
  
  cupsFreeDests(amount, dests);
  return 0;
}
```

**Comment by michaelrsweet** [comment]
@SonnyX Two things - first, if you bother to connect to the destination, you should use that connection when calling cupsCopyDestInfo.

Second, are these existing print queues that show up with "lpstat -v" or are they just printers on the network that haven't been added? It *should* work either way but will help me focus my efforts on figuring out what is going wrong.

**Comment by SonnyX** [comment]
To the first issue: I tried both, so I did use that connection, and I also require that connection to print to the actual printers, I just made example code to reflect and emulate my issue in a short fashion.

To the second question: these are non-existing queue's and thus with the cupsConnectDest I create new temporary queues.

My operating system is `Debian`, I got the `cups` package from the `sid` distribution.
My `uname -a`: `Linux Hostname-Omitted 4.19.0-16-amd64 #1 SMP Debian 4.19.181-1 (2021-03-19) x86_64 GNU/Linux`
